src/utility.ts

```diff
@@ -109,3 +113,25 @@ export function formatErrorProperties(ex: any): IProperties {
 
     return properties;
 }
+
+export function checkJavaHome(): Promise<string> {
+    const EXE_SUFFIX = process.platform.startsWith("win") ? ".exe" : "";
+    return new Promise((resolve, reject) => {
+        let javaHome: string = readJavaConfig() || process.env.JDK_HOME || process.env.JAVA_HOME;
+        if (javaHome) {
+            javaHome = expandHomeDir(javaHome);
+            if (pathExists.sync(javaHome) && pathExists.sync(path.resolve(javaHome, "bin", `javac${EXE_SUFFIX}`))) {
+                return resolve(javaHome);
+            }
+        }
+
+        findJavaHome((err, home) => {

```

Comment:
will findJavaHome expand the home dir like above expandHomeDir?

Reply:
nice catch.